[Javascript] Implement boolean-logic concept exercise (#841)
exercism/v3: languages/javascript/exercises/concept/booleans/.docs/hints.md

### 1. Check if the 'Fast Attack' action is possible

- The logical NOT operator (`!`) can be placed before an expression to negate its value.

### 2. Check if the 'Spy' action is possible

> - Logical expressions are evaluated from left to right and are tested for possible 'short-circuits'

Reviewer: ```suggestion
- Logical expressions are evaluated from left to right and are tested for possible 'short-circuits'.
```

### 3. Check if the 'Signal Prisoner' action is possible

> - Logical Operators in the order of their precedence (from highest to lowest): `!`, `&&`, `||`

Reviewer: ```suggestion
- Logical Operators in the order of their precedence (from highest to lowest): `!`, `&&`, `||`.
```

# Hints

### 1. Decide if 'Fast Attack' action is possible

> - The logical NOT operator (!) is placed before an expression

Reviewer: ```suggestion
- The logical NOT operator (!) can be placed before an expression to negate its value.
```

### 2. Decide if 'Approach and Spy' action is possible

- Logical expressions are evaluated from left to right and are tested for possible "short-circuits"

### 3. Decide if 'Signal Prisoner' action is possible

> - Logical Operators in the order of their precedence (from highest to lowest): !, &&, ||

Reviewer: ```suggestion
- Logical Operators in the order of their precedence (from highest to lowest): `!`, `&&`, `||`
```
